storage/src/adb/store/mod.rs

```diff
@@ -202,6 +205,123 @@ pub trait Db<K: Array, V: Codec> {
     fn destroy(self) -> impl Future<Output = Result<(), Error>>;
 }
 
+pub struct Batch<'a, K, V, D>
+where
+    K: Array,
+    V: Codec + Clone,
+    D: Batchable<K, V>,
+{
+    db: &'a mut D,
+    diff: BTreeMap<K, Option<V>>,
```

Comment:
None of the (A)DBs are no_std so might be better to use HashMap unless we can think of a reason why the ordering helps.